### PR TITLE
Change unit tests to avoid making mypy a main requirement.

### LINKED ISSUE #167: Update unit test so installation works without mypy
Currently `pip install -e .` fails as it need mypy not in `requirements/main.txt`. Update unit test with mypy check or mark as 'dev'.

### PR DESCRIPTION
**Pull request by luisenp**
## Motivation and Context
Closes #167

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,22 +25,27 @@ The current focus is on nonlinear least squares with support for sparsity, batch
     git clone https://github.com/facebookresearch/theseus.git && cd theseus
     pip install -e .
     ```
+    If you are interested in contributing to `theseus`, also install
```
Comment (mhmukadam) [inline README.md:28]: also -> instead?

File: README.md [hunk 2]
```diff
@@ -25,22 +25,27 @@ The current focus is on nonlinear least squares with support for sparsity, batch
...
+    ```
+    and follow the more detailed instructions in [CONTRIBUTING](https://github.com/facebookresearch/theseus/blob/main/CONTRIBUTING.md).
 - Running unit tests
     ```bash
     pytest theseus
     ```
   By default, unit tests include tests for our CUDA extensions. You can add the option `-m "not cudaext"` to 
   skip them when installing without CUDA support.
-- See [tutorials](tutorials/) and [examples](examples/) to learn about the API and usage.
+- See [tutorials](https://github.com/facebookresearch/theseus/blob/main/tutorials/) and [examples](https://github.com/facebookresearch/theseus/blob/main/examples/) to learn about the API and usage.
```
Comment (mhmukadam) [inline README.md:39]: Do the relative links not work on your end? For me they work on github as well as VS code.

File: .circleci/config.yml [hunk 3]
```diff
@@ -110,7 +110,6 @@ setup_project: &setup_project
       working_directory: ~/project
       command: |
         pip install --progress-bar off -e .
-        pip install -r requirements/dev.txt
```
Comment (mhmukadam) [inline .circleci/config.yml:113]: Does this mean the mypy related test won't execute on CI?

File: README.md (hunk 2 above)
Comment (luisenp) [reply README.md:39]: They work but they won't work in `pypi`'s landing page.

**Review by mhmukadam — APPROVED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (luisenp) [reply .circleci/config.yml:113]: Yes. I was typing an answer as to why I think this was OK, but I actually changed my mind in the middle of writing it lol. I'll add this back.

File: README.md (hunk 1 above)
Comment (luisenp) [reply README.md]: Changed.

**Pull request merged**